substratevm/src/com.oracle.svm.jni/src/com/oracle/svm/jni/hosted/JNICallTrampolineMethod.java

```diff
@@ -130,7 +130,7 @@ public CompileFunction createCustomCompileFunction() {
             CallingConvention callingConvention = backend.getCodeCache().getRegisterConfig().getCallingConvention(
                     SubstrateCallingConventionKind.Native.toType(true), returnType, parameters.toArray(new JavaType[0]), backend);
 
-            return backend.createJNITrampolineMethod(method, identifier, threadIsolateOffset, nonVirtual, getFieldOffset(providers), callingConvention);
+            return backend.createJNITrampolineMethod(method, identifier, 0, threadIsolateOffset, nonVirtual ? 3 : 2, getFieldOffset(providers), callingConvention);
```

Comment:
Again, these numbers should come from somewhere, in that case the `parameters` list that is being built above.